Update note for breaking change in Postgres 17.1. (#7481)
timescale/timescaledb: README.md

### Installation

Installation options are:

> - **[Timescale Cloud](https://tsdb.co/GitHubTimescale)**: a fully-managed TimescaleDB in the cloud, is

Reviewer: ```suggestion
- **[Timescale Cloud](https://tsdb.co/GitHubTimescale)**: A fully-managed TimescaleDB in the cloud, is
```

  available via a free trial. Create a PostgreSQL database in the cloud with TimescaleDB pre-installed
  so you can power your application with TimescaleDB without the management overhead.

- **Platform packages**: TimescaleDB is also available pre-packaged for several platforms such as
  Linux, Windows, MacOS, Docker, and Kubernetes. For more information, see [Install TimescaleDB](https://docs.timescale.com/self-hosted/latest/install/).

> - **Build from source**: see [Building from source](https://github.com/timescale/timescaledb/blob/main/docs/BuildSource.md).

Reviewer: ```suggestion
- **Build from source**: See [Building from source](https://github.com/timescale/timescaledb/blob/main/docs/BuildSource.md).
```

>   PostgreSQL 17.1 [introduced a breaking change][postgres-breaking-change] that is incompatible with TimescaleDB. When you
>   build from source, best practice is to build with PostgreSQL 17.2 and higher. Do not build against PostgreSQL 17.1.
>   Users of  [Timescale Cloud](https://console.cloud.timescale.com/) and Platform packages are unaffected

Reviewer: ```suggestion
  We recommend not using TimescaleDB with PostgreSQL 17.1, 16.5, 15.9, 14.14, 13.17, 12.21.  
  These minor versions [introduced a breaking binary interface change][postgres-breaking-change] that, 
  once identified, was reverted in subsequent minor PostgreSQL versions 17.2, 16.6, 15.10, 14.15, 13.18, and 12.22.
  When you build from source, best practice is to build with PostgreSQL 17.2, 16.6, etc and higher. 
  Users of [Timescale Cloud](https://console.cloud.timescale.com/) and Platform packages built and 
  distributed by Timescale are unaffected.
```
